phonemizer/punctuation.py

```diff
@@ -60,14 +60,19 @@ def marks(self):
         return self._marks
 
     @marks.setter
-    def marks(self, value: str):
-        if not isinstance(value, str):
-            raise ValueError('punctuation marks must be defined as a string')
-        self._marks = ''.join(set(value))
-
-        # catching all the marks in one regular expression: zero or more spaces
-        # + one or more marks + zero or more spaces.
-        self._marks_re = re.compile(fr'(\s*[{re.escape(self._marks)}]+\s*)+')
+    def marks(self, value: Union[str, re.Pattern]):
+        if isinstance(value, re.Pattern):
+            # catch the pattern surrounded by zero or more spaces on either side
+            self._marks_re = re.compile(r'\s*(' + value.pattern + r')+\s*')
```

Comment:
All of the rest of the code assumes the adjacent spaces are captured as part of the punctuation. But I thought it might be better not to require the end users to match them.